File: Cs/Solution/SumRange.cs
using System;

namespace Solution
{
	public class NumArray
	{
		private int[] nums;

		public NumArray(int[] nums)
		{
			this.nums = nums;
		}

		public int SumRange(int left, int right)
		{
			int sum = 0;
			for (int i = left; i <= right; i++)
			{
				sum += nums[i];
			}
			return sum;
		}
	}
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.SumRange(left,right);
 */

File: Ruby/solution/sum_range.rb
class NumArray

=begin
    :type nums: Integer[]
=end
  def initialize(nums)
    @nums = nums
  end

=begin
    :type left: Integer
    :type right: Integer
    :rtype: Integer
=end
  def sum_range(left, right)
    @nums[left..right].sum
  end
end

# Your NumArray object will be instantiated and called as such:
# obj = NumArray.new(nums)
# param_1 = obj.sum_range(left, right)
